meshtastic/node_runtime/content_runtime.py

```
@@ -0,0 +1,436 @@
+        read_generation = begin_read_generation()
+        response_event = threading.Event()
+
+        def _on_response(packet: dict[str, Any]) -> None:
+            if not is_read_generation_active(read_generation):
+                logger.debug(
+                    "Ignoring late content response for retired read generation %s",
+                    read_generation,
+                )
+                return
+            if handle_response(packet):
+                if is_read_generation_active(read_generation):
```

Ruff (0.15.6)

[warning] Use a single `if` statement instead of nested `if` statements

(SIM102)